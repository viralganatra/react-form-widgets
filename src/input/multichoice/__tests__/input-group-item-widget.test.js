import React from 'react';
import InputGroupItemWidget from '../input-group-item-widget';

jest.mock('../../single/input');
jest.mock('../../../label/label-widget');

const data = {
    label: {
        content: 'Label',
        className: 'label-class',
    },
    type: 'checkbox',
    value: '2',
    checked: true,
};

describe('InputGroupItemWidget', () => {
    it('should render correctly', () => {
        const methodSpy = jest.fn();
        const wrapper = shallow(
            <InputGroupItemWidget {...data} onChange={methodSpy} />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should validate prop types', () => {
        const error = console.error;
        console.error = jest.fn();

        shallow(<InputGroupItemWidget label="test" value="2" />);
        shallow(<InputGroupItemWidget {...data} value={2} />);

        expect(console.error.mock.calls[0][0]).toBeDefined();
        expect(console.error.mock.calls[1][0]).toBeDefined();

        console.error = error;
    });
});
