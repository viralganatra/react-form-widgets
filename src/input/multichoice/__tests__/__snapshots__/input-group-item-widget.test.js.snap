exports[`InputGroupItemWidget should render correctly 1`] = `
<Label
  className="label-class">
  <Input
    checked={true}
    onChange={[Function]}
    type="checkbox"
    value="2" />
   
  Label
</Label>
`;
